mesonbuild/interpreter/interpreter.py

```diff
@@ -617,6 +617,9 @@ def func_import(self, node: mparser.BaseNode, args: T.Tuple[str],
         # Some tests use "unstable_" instead of "unstable-", and that happens to work because
         # of implementation details
         if modname.startswith('unstable'):
+            if modname.startswith('unstable_'):
+                mlog.deprecation(f'Importing unstable modules as "{modname}" instead of "{modname.replace("_", "-", 1)}"',
+                                 location=node)
             modname = modname[len('unstable') + 1:]  # + 1 to handle the - or _
             expect_unstable = True
```

Comment:
Doesn't this match anything that starts with `unstable`, including `unstable` and `unstableee`?

Reply:
Yeah, it would, though we don't have any such modules

Reply:
Sure, but because of the name mangling (removal of the prefix) wouldn't the error message be a bit surprising?

Reply:
```python
if modname.replace('_', '-', 1).startswith('unstable-'):
    if modname.startswith('unstable_'):
        ...
    ...
```

Reply:
I've reworked it in the latest version.